components/table/src/cell/cell-fixed.directive.ts

```diff
@@ -50,11 +36,27 @@ export class NzCellFixedDirective implements OnChanges {
     this.renderer.setStyle(this.elementRef.nativeElement, 'right', autoRight);
   }
 
-  constructor(private cdr: ChangeDetectorRef, private renderer: Renderer2, private elementRef: ElementRef) {}
+  setIsFirstRight(isRightFirst: boolean): void {
+    this.setFixClass(isRightFirst, 'ant-table-cell-fix-right-first');
+  }
+
+  setIsLastLeft(isLeftLast: boolean): void {
+    this.setFixClass(isLeftLast, 'ant-table-cell-fix-left-last');
+  }
+
+  private setFixClass(flag: boolean, className: string): void {
```

Comment:
should remove className whenever the flag is true, since the setFixClass function may call many times.